### PR TITLE
[tests] add the missing `require_sacremoses` decorator 

### PR DESCRIPTION
**Pull request by faaany**
## What does this PR do?
When running:
```bash
python -m pytest -rA tests/models/biogpt
```
I got a lot of failed tests due to uninstalled package:
```bash
FAILED tests/models/biogpt/test_tokenization_biogpt.py::BioGptTokenizationTest::test_special_tokens_mask_input_pairs - ImportError: You need to install sacremoses to use BioGptTokenizer. See https://pypi.org/project/sacremoses/ for installation.
FAILED tests/models/biogpt/test_tokenization_biogpt.py::BioGptTokenizationTest::test_token_type_ids - ImportError: You need to install sacremoses to use BioGptTokenizer. See https://pypi.org/project/sacremoses/ for installation.
FAILED tests/models/biogpt/test_tokenization_biogpt.py::BioGptTokenizationTest::test_tokenize_special_tokens - ImportError: You need to install sacremoses to use BioGptTokenizer. See https://pypi.org/project/sacremoses/ for installation.
FAILED tests/models/biogpt/test_tokenization_biogpt.py::BioGptTokenizationTest::test_tokenizer_slow_store_full_signature - ImportError: You need to install sacremoses to use BioGptTokenizer. See https://pypi.org/project/sacremoses/ for installation.
FAILED tests/models/biogpt/test_tokenization_biogpt.py::BioGptTokenizationTest::test_tokenizers_common_ids_setters - ImportError: You need to install sacremoses to use BioGptTokenizer. See https://pypi.org/project/sacremoses/ for installation.
FAILED tests/models/biogpt/test_tokenization_biogpt.py::BioGptTokenizationTest::test_tokenizers_common_properties - ImportError: You need to install sacremoses to use BioGptTokenizer. See https://pypi.org/project/sacremoses/ for installation.
FAILED tests/models/biogpt/test_tokenization_biogpt.py::BioGptTokenizationTest::test_tokenizers_special_tokens_properties_unset_0 - ImportError: You need to install sacremoses to use BioGptTokenizer. See https://pypi.org/project/sacremoses/ for installation.
FAILED tests/models/biogpt/test_tokenization_biogpt.py::BioGptTokenizationTest::test_tokenizers_special_tokens_properties_unset_1 - ImportError: You need to install sacremoses to use BioGptTokenizer. See https://pypi.org/project/sacremoses/ for installation.
=============================== 49 failed, 132 passed, 57 skipped, 19 warnings in 85.18s (0:01:25) ===============================
```
The reason lies at [this](https://github.com/huggingface/transformers/blob/d45f47ab7f7c31991bb98a0302ded59ab6adac31/src/transformers/models/biogpt/tokenization_biogpt.py#L118) line. Instead of letting tests fail, we should add a decorator to skip these tests if the package is not installed.

### CODE REVIEW

File: src/transformers/testing_utils.py [hunk 1]
```diff
@@ -562,6 +563,13 @@ def require_sentencepiece(test_case):
     return unittest.skipUnless(is_sentencepiece_available(), "test requires SentencePiece")(test_case)
 
 
+def require_sacremoses(test_case):
+    """
+    Decorator marking a test that requires Sacremoses. These tests are skipped when SentencePiece isn't installed.
```
Comment (amyeroberts) [inline src/transformers/testing_utils.py:568]:
```suggestion
    Decorator marking a test that requires Sacremoses. These tests are skipped when Sacremoses isn't installed.
```

**Review by amyeroberts — APPROVED** [review]
LGTM - thanks for adding! 

Just a small nit

File: src/transformers/testing_utils.py (hunk 1 above)
Comment (faaany) [reply src/transformers/testing_utils.py:568]: good catch! thanks so much!

**Comment by amyeroberts** [comment]
Hi @faaany, there's a failing test, which I don't think is related to this PR but I haven't seen on other PRs. Could you try rebasing onto main to see if this resolves the issue?

**Comment by faaany** [comment]
> Hi @faaany, there's a failing test, which I don't think is related to this PR but I haven't seen on other PRs. Could you try rebasing onto main to see if this resolves the issue?

done and yes, the failed test is not related to this PR.

**Pull request merged**